### PR TITLE
Adding access control implementation.(AT-4)

### PR DESCRIPTION
**Pull request by hweng**

### CODE REVIEW

File: app/controllers/Hydra/Shibboleth/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module Hydra
```
Comment (jcoyne) [inline app/controllers/Hydra/Shibboleth/application_controller.rb]: The directories should be all lower cased

File: app/models/Hydra/Shibboleth/user.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+require 'net/ldap'
```
Comment (jcoyne) [inline app/models/Hydra/Shibboleth/user.rb]: The directory names should be all lower case.

File: db/migrate/20150924230240_create_hydra_shibboleth_user.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class CreateHydraShibbolethUser < ActiveRecord::Migration
+  def change
+    create_table :hydra_shibboleth_users do |t|
+      t.string :email,              :null => true, :default => ""
```
Comment (jcoyne) [inline db/migrate/20150924230240_create_hydra_shibboleth_user.rb:4]: Why give empty strings as default if null is true?

File: app/models/Hydra/Shibboleth/user.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        ldap.search(:filter => composite_filter, :attributes => result_attrs, :return_result => false) { |item| 
+           result = item.sAMAccountName.first}
+
+        get_ldap_response(ldap)
+
+        return result
+      end
+
+      def self.get_ldap_connection
+        ldap = Net::LDAP.new  :host => Rails.application.secrets.ldap_host,
```
Comment (jcoyne) [inline app/models/Hydra/Shibboleth/user.rb]: I'd suggest putting this in a configuration object which can read from Rails.application.secrets. In this way someone could swap in a different configuration that reads from elsewhere such as ENV variables.

**Comment by hweng** [comment]
The .DS_Store has been put in my global gitignore, and removed from project gitignore. The case of dir was fixed. Thanks!

File: db/migrate/20150924230240_create_hydra_shibboleth_user.rb (hunk 3 above)
Comment (hweng) [inline db/migrate/20150924230240_create_hydra_shibboleth_user.rb:4]: I think maybe it doesn't matter if give empty strings as default or not, there would be slight difference when check those fields using nil? vs. empty?

**Comment by hweng** [comment]
Agree. And I'll refactor it to use hydra-ldap instead for LDAP group lookup since this would let us build on an existing community project and have more options.

**Comment by hweng** [comment]
Changes merge into https://github.com/projecthydra-labs/hydra-shibboleth/pull/2

**Pull request closed without merging**